buffer.c

```diff
@@ -835,29 +741,29 @@ PHP_FUNCTION(array_buffer_view_wakeup)
 	}
 
 	zend_throw_exception(
-		NULL, "Invalid serialization data", 0 TSRMLS_CC
+		NULL, "Invalid serialization data", 0
 	);
 }
 
 PHP_FUNCTION(array_buffer_view_offset_get)
 {
 	buffer_view_object *intern;
 	long offset;
-	zval *retval;
+	zval retval;
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "l", &offset) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "l", &offset) == FAILURE) {
 		return;
 	}
 
-	intern = zend_object_store_get_object(getThis() TSRMLS_CC);
+	intern = Z_BUFFER_VIEW_OBJ_P(getThis());
 
 	if (offset < 0 || offset >= intern->length) {
-		zend_throw_exception(NULL, "Offset is outside the buffer range", 0 TSRMLS_CC);
+		zend_throw_exception(NULL, "Offset is outside the buffer range", 0);
 		return;
 	}
 
-	retval = buffer_view_offset_get(intern, offset);
-	RETURN_ZVAL(retval, 1, 1);
+	buffer_view_offset_get(intern, offset, &retval);
```

Comment:
Can directory write into `return_value` here.

Reply:
Sorry, I don't understand what you mean. Write directly into `return_value` how?

This is the current code:

```c
	buffer_view_offset_get(intern, offset, &retval);
	RETURN_ZVAL(&retval, 1, 0);
```

I gather the `RETURN_ZVAL` can be elided in some way...?

Reply:
It's possible to write `buffer_view_offset_get(intern, offset, return_value);`